types/wlr_output_layout.c

```diff
@@ -190,6 +229,39 @@ static struct wlr_output_layout_output *output_layout_output_create(
 	return l_output;
 }
 
+void relocate_output_with_children(struct wlr_output_layout *layout,
+		struct wlr_output_layout_output *output,
+		struct wl_list *ref) {
+
+	struct wlr_output_layout_output *child_output;
+	struct wlr_output_layout_output *parent_output =
+		output;
+
+	if (output->link.prev == ref) {
+		return;
+	}
+
+	for (child_output = wl_container_of((output)->link.next, output, link);
```

Comment:
Not a fan of this complicated loop

Reply:
Not sure how to do it with `wl_list_for_each`, I need to skip to the output that's being moved. It could be possible, but I don't feel it is all that cleaner.